src/Symfony/Bridge/Doctrine/ArgumentResolver/UserInjector.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\ArgumentResolver;
+
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Default variable loader service used to inject user in EntityValueResolver.
+ *
+ * @author Roman JOLY <eltharin18@outlook.fr>
+ */
+class UserInjector implements EntityValueResolverVariableInjectorInterface
+{
+    public function __construct(private TokenStorageInterface $tokenStorage)
+    {
+    }
+
+    public function getVariables(): array
+    {
+        return ['user' => $this->tokenStorage?->getToken()?->getUser()];
```

Comment:
the way this is implemented means that all usages of an expression to resolve the entity will make the security token as being used (and so the session as being used for non-stateless firewalls) even when the expression does not care about the variable. this is bad for the compatibility with the session usage tracking (especially if there is no way to disable this UserInjector entirely as a workaround)

Reply:
a way for disable UserInjector is to replace service without it (I agree it's not very good)

many ideas (I haven’t yet thought about whether this is feasible.) : 
- add a security config option to enable/disable UserInjector by default
- add an argument in mapEntity Attribute to specify an EntityValueResolverVariableInjector for this specific Resolver